### ISSUE TITLE
get rid of "The order in which module records have their [[AsyncEvaluation]] fields transition to true is significant"

### ISSUE THREAD

**Issue by bakkot** [tc39/ecma262#3289]
[`AsyncModuleExecutionFulfilled`](https://tc39.es/ecma262/multipage/ecmascript-language-scripts-and-modules.html#sec-async-module-execution-fulfilled) sorts a list according to "the order in which [the elements] had their [[AsyncEvaluation]] fields set to `true`".

That's pretty wild: it implies you need to keep track of not just the current state of the spec's data structures, but the order in which they changed. We should get rid of that.

The simplest fix would be to have a global integer which is incremented every time we use it, and store that in the field instead of the boolean.

**Comment by syg** [comment]
Indeed, the global integer (modulo some overflow handling complexity that's not needed in the spec) is how V8 implements that logic anyhow.
